pkg/scheduler/plugins/loadaware/pod_assign_cache.go

```diff
@@ -211,3 +349,294 @@ func (p *podAssignCache) OnDelete(obj interface{}) {
 	}
 	p.unAssign(pod.Spec.NodeName, pod)
 }
+
+// AddOrUpdatePod add or update pod to nodeInfo.
+// It returns false is nodeInfo is already deleted and caller should get a new nodeInfo and retry.
+// Unlock is called whether nodeInfo is locked before calling or not.
+func (n *nodeInfo) AddOrUpdatePod(pod *podAssignInfo, locked bool) bool {
+	if n.deleted {
```

Comment:
If the node metric is deleted, the filtering of the node should either reject the scheduling pod or skip the check (determined by the plugin args), neither of which needs the assigned pod. So I suppose we can abort the pod event, and let the nodeInfo be missing until the node metric comes back.